include/zephyr/drivers/comparator/nrf_lpcomp.h

```diff
@@ -84,9 +56,9 @@ enum comp_nrf_lpcomp_refsel {
  */
 struct comp_nrf_lpcomp_config {
 	/** Positive input selection */
-	enum comp_nrf_lpcomp_psel psel;
+	uint8_t psel;
```

Comment:
This change requires including the bindings which define the psel and refsel, and extending the documentation both the defines in the bindings, similar to the comments in the enums. So, add
```
/** AIN0 external input */
```
the the defines in the bindings file and extend the psel docs to mention where the defines are from
```
/** Positive input selection  defined by the NRF_LPCOMP_ defines */
```
and add the required header above to the header file instead if the source file as its required by the user to use the API
```
#include <zephyr/dt-bindings/comparator/nrf-lpcomp.h>
```

Reply:
Since we loose the enums, we loose the strict types, so we need to be extra careful to document exactly what integer goes where, since anything could be passed to the API